platform/jobs/edxPlatformBokChoyMaster.groovy

```diff
@@ -0,0 +1,178 @@
+package devops
+
+import hudson.model.Build
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_LOG_ROTATOR
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_WORKER
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_PARSE_SECRET
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_HIPCHAT
+import static org.edx.jenkins.dsl.JenkinsPublicConstants.JENKINS_PUBLIC_BASE_URL
+
+/*
+Example secret YAML file used by this script
+publicJobConfig:
+    open : true/false
+    jobName : name-of-jenkins-job-to-be
+    url: github-url-segment
+    url-clone : github-url-segment-clone
+    credential : n/a
+    credential-clone : n/a
+    cloneReference : clone/.git
+    email : email-address@email.com
+    hipchat : token
+*/
+
+Map <String, String> predefinedPropsMap  = [:]
+predefinedPropsMap.put('GIT_SHA', '${GIT_COMMIT}')
+predefinedPropsMap.put('GITHUB_ORG', 'edx')
+predefinedPropsMap.put('GITHUB_REPO', 'edx-platform')
+predefinedPropsMap.put('TARGET_URL', JENKINS_PUBLIC_BASE_URL + 'job/edx-platform-bok-choy-master/${BUILD_NUMBER}/')
+predefinedPropsMap.put('CONTEXT', 'jenkins/bokchoy')
+
+String jUnitReports = 'edx-platform/**/nosetests.xml,edx-platform/reports/acceptance/*.xml,'
+jUnitReports += 'edx-platform/reports/quality.xml,edx-platform/reports/javascript/javascript_xunit.xml,'
+jUnitReports += 'edx-platform/reports/bok_choy/xunit.xml,edx-platform/reports/bok_choy/**/xunit.xml'
+
+/* stdout logger */
+/* use this instead of println, because you can pass it into closures or other scripts. */
+/* TODO: Move this into JenkinsPublicConstants, as it can be shared. */
+Map config = [:]
```

Comment:
I think we might have at least 3 job definitions that now have this copy/pasted code. It probably would be a good time to refactor and pull this out to a shared location.